src/main/java/de/hysky/skyblocker/skyblock/item/slottext/adders/EnchantmentLevelAdder.java

```diff
@@ -1,21 +1,201 @@
+		ID_ABBREVIATION_MAP.put("power", "PW");
+		ID_ABBREVIATION_MAP.put("pristine", "PM"); // Prismatic
+		ID_ABBREVIATION_MAP.put("projectile_protection", "PP");
+		ID_ABBREVIATION_MAP.put("prosecute", "PO");
+		ID_ABBREVIATION_MAP.put("prosperity", "PE");
+		ID_ABBREVIATION_MAP.put("protection", "PR");
+		ID_ABBREVIATION_MAP.put("punch", "PU");
+
+		ID_ABBREVIATION_MAP.put("quantum", "QU");
+
+		ID_ABBREVIATION_MAP.put("rainbow", "RA");
+		ID_ABBREVIATION_MAP.put("reflection", "RF");
+		ID_ABBREVIATION_MAP.put("rejuvenate", "RJ");
+		ID_ABBREVIATION_MAP.put("replenish", "RP");
+		ID_ABBREVIATION_MAP.put("respiration", "RE");
+		ID_ABBREVIATION_MAP.put("respite", "RS");
+
+		ID_ABBREVIATION_MAP.put("scavenger", "SC");
+		ID_ABBREVIATION_MAP.put("sharpness", "SH");
+		ID_ABBREVIATION_MAP.put("silk_touch", "ST");
+		ID_ABBREVIATION_MAP.put("smarty_pants", "SP");
+		ID_ABBREVIATION_MAP.put("smelting_touch", "SE");
+		ID_ABBREVIATION_MAP.put("smite", "SI");
+		ID_ABBREVIATION_MAP.put("smoldering", "SL");
+		ID_ABBREVIATION_MAP.put("snipe", "SN");
+		ID_ABBREVIATION_MAP.put("spiked_hook", "SK");
+		ID_ABBREVIATION_MAP.put("strong_mana", "SM");
+		ID_ABBREVIATION_MAP.put("sugar_rush", "SR");
+		ID_ABBREVIATION_MAP.put("sunder", "SU");
+		ID_ABBREVIATION_MAP.put("syphon", "SY");
+
+		ID_ABBREVIATION_MAP.put("tabasco", "TA");
+		ID_ABBREVIATION_MAP.put("thorns", "TN");
+		ID_ABBREVIATION_MAP.put("thunderbolt", "TB");
+		ID_ABBREVIATION_MAP.put("thunderlord", "TL");
+		ID_ABBREVIATION_MAP.put("titan_killer", "TK");
+		ID_ABBREVIATION_MAP.put("toxophilite", "TX");
+		ID_ABBREVIATION_MAP.put("transylvanian", "TY");
+		ID_ABBREVIATION_MAP.put("triple_strike", "TS");
+		ID_ABBREVIATION_MAP.put("true_protection", "TP");
+
+		// Turbo books
+		ID_ABBREVIATION_MAP.put("turbo_cacti", "TI");
+		ID_ABBREVIATION_MAP.put("turbo_cane", "TC");
+		ID_ABBREVIATION_MAP.put("turbo_carrot", "TR");
+		ID_ABBREVIATION_MAP.put("turbo_coco", "TO");
+		ID_ABBREVIATION_MAP.put("turbo_melon", "TE");
+		ID_ABBREVIATION_MAP.put("turbo_mushrooms", "TM");
+		ID_ABBREVIATION_MAP.put("turbo_potato", "TT");
+		ID_ABBREVIATION_MAP.put("turbo_pumpkin", "TU");
+		ID_ABBREVIATION_MAP.put("turbo_warts", "TW");
+		ID_ABBREVIATION_MAP.put("turbo_wheat", "TH");
+
+		ID_ABBREVIATION_MAP.put("vampirism", "VP");
+		ID_ABBREVIATION_MAP.put("venomous", "VE");
+		ID_ABBREVIATION_MAP.put("vicious", "VI");
+
+		// check if all abbreviations are unique, if not report duplicates
+		if (Debug.debugEnabled() && ID_ABBREVIATION_MAP.size() != (int) ID_ABBREVIATION_MAP.values().stream().distinct().count()) {
+			Map<String, String> otherDirection = new Object2ObjectOpenHashMap<>();
+			for (Map.Entry<String, String> entry : ID_ABBREVIATION_MAP.entrySet()) {
+				final String put = otherDirection.put(entry.getValue(), entry.getKey());
+				if (put != null) {
+					throw new IllegalArgumentException("Duplicate key [%s, %s] for value %s".formatted(put, entry.getKey(), entry.getValue()));
```

Comment:
No need to throw here, logging an error is sufficient.

Reply:
I explicitly decided to throw an exception, since you may not notice a simple log. And since the exception will only be throw in a debug/development environment I assumed it should be fine

Reply:
Actually, this would be more sufficient as a unit test. This way we also don't ship test code.